feat: LEAP-1099: Allow to update annotation only after changes

hlomzik opened the pull request:
With FF ON the Update button will be disabled if there were no changes made to annotation. Also Update button will always say "Update" to fix some legacy leftover.

### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated
- [ ] Docs have been added/updated
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance

### Describe the reason for change
An experiment.

### What feature flags were used to cover this change?
`fflag_feat_all_leap_1081_reviewer_flow_updates`

### This change affects (describe how if yes)
- [ ] Performance
- [ ] Security
- [x] UX

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No - to update annotation no mater what users can just add any region and remove it
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [ ] e2e (codecept)
- [ ] integration (cypress)
- [ ] unit (jest)

hlomzik committed:
Add FF for new Reviewer Flow

hlomzik committed:
feat: LEAP-1099: Allow to update annotation only with changes

With FF on Update button will be disabled if there were no changes made to annotation.
Also Update button will always say "Update" to fix some legacy leftover.

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9290855881

bmartel commented on web/libs/editor/src/components/BottomBar/Controls.jsx:
```diff
@@ -245,15 +245,17 @@ export const Controls = controlsInjector(
+          const isUpdateDisabled = isDisabled || noChanges;
           const button = (
-            <ButtonTooltip key="update" title="Update this task: [ Alt+Enter ]">
+            <ButtonTooltip key="update" title={noChanges ? "No changes were made" : "Update this task: [ Alt+Enter ]"}>
```
```
            <ButtonTooltip key="update" title={noChanges ? "No changes were made" : "Update this task: [ Alt+Enter ]"}>
```
Do these changes to disable the Button also disable the hotkey submission as well?

hlomzik replied:
With `noChanges` button effectively be disabled, so I think that's okay to not show hotkey at this moment, because it will be useless

hlomzik committed:
Fix hotkeys as well

Check for changes before update in hotkey handler

hlomzik committed:
Add comments to different Controls components

Mostly to deprecate anything except the new interface.

hlomzik committed:
Add missed import

hlomzik committed:
Merge branch 'fb-leap-1099/update-only-changed' of github.com:heartexlabs/label-studio into fb-leap-1099/update-only-changed

hlomzik committed:
Don't block Update if there is a draft saved previously

It's a change over the last submitted annotation, so Update should be available.

hlomzik committed:
Fix hotkey in tooltip

We have the same hotkey for both Submit, Update and Approve: ctrl+enter/cmd+enter.
alt+enter is for Skip (and most probably it doesn't work because it's set for Mac).
So while "Ctrl+Enter" is for Windows only at least it's not 100% wrong.
See https://github.com/HumanSignal/label-studio/blob/2660ec98/web/libs/editor/src/core/settings/keymap.json#L36-L40

hlomzik committed:
Merge branch 'develop' into fb-leap-1099/update-only-changed

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9357788912
